html/modules/custom/reliefweb_reporting/src/Commands/ReliefWebReportingCommands.php

```diff
@@ -760,6 +776,52 @@ public function exportReportData(
       fclose($file);
     }
 
+    if ($upload) {
+      $client = new Client();
+      if ($credentials = getenv('GOOGLE_APPLICATION_CREDENTIALS')) {
+        // Suppress error to avoid echoing the credential in a traceback.
+        if (!@file_exists($credentials)) {
+          $client->setAuthConfig($credentials);
+        }
+        else {
+          $client->useApplicationDefaultCredentials();
+        }
+      }
+      else {
+        $this->logger->error('Error: No credentials defined in the GOOGLE_APPLICATION_CREDENTIALS environment.');
+        return FALSE;
+      }
+
+      $client->setApplicationName("Reliefweb Reports Data Uploader");
+      $client->setScopes(['https://www.googleapis.com/auth/drive']);
+      $service = new Drive($client);
+
+      $file = new DriveFile();
+      $file->setName(basename($output));
+      $file->setParents([$options['gdrive-upload-folder']]);
+
+      try {
+        $result = $service->files->create($file, [
+          'data' => file_get_contents($output),
```

Comment:
Note: the files can very large. If we encounter issues, we'll have to stream the file and send chunks.